Mu2eG4/src/constructDiskCalorimeter.cc

```diff
@@ -289,48 +297,114 @@ namespace mu2e {
        reg.add(rotFPPipe);
        
        G4Torus*         coolFP     = new G4Torus("caloCoolFP",FPCoolPipeRadius-FPCoolPipeThickness, FPCoolPipeRadius, FPCoolPipeTorRadius, angMax, CLHEP::twopi-2.0*angMax);
-       G4LogicalVolume* coolFPLog  = caloBuildLogical(coolFP, pipeMaterial, "caloCoolFPLog_"+std::to_string(idisk),isPipeVisible,G4Color::Red(),isPipeSolid,0);
+       G4LogicalVolume* coolFPLog  = caloBuildLogical(coolFP, pipeMaterial, "caloCoolFPLog"+std::to_string(idisk),isPipeVisible,G4Color::Red(),isPipeSolid,0);
        pv = new G4PVPlacement(rotFPPipe,G4ThreeVector(0.0,0.0,ZposPipe), coolFPLog, "caloCoolFPPV", frontPlateLog, false, 0, false);
        doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);                
        
 
        //pipes nside foam
-       G4RotationMatrix* rotPipe1 = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipe1->rotateZ(CLHEP::pi/2.0);
-       G4RotationMatrix* rotPipe2 = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipe2->rotateZ(1.5*CLHEP::pi);
-       G4RotationMatrix* rotPipeFlat = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipeFlat->rotateX(CLHEP::pi/2.0);
+       G4RotationMatrix* rotPipe1 = new G4RotationMatrix();
+       //rotPipe1->rotateZ(CLHEP::pi/2.0);
+       rotPipe1->rotateZ(0.0 * CLHEP::pi);
        reg.add(rotPipe1);
+       G4RotationMatrix* rotPipe2 = new G4RotationMatrix();
+       rotPipe2->rotateZ(1.0 * CLHEP::pi); 
        reg.add(rotPipe2);
```

Comment:
The one liners using e.g.,:
G4RotationMatrix* rotPipe2 = reg.add(new G4RotationMatrix());
are preferred over 
G4RotationMatrix* rotPipe2 = new G4RotationMatrix();
reg.add(rotPipe2);
as they are easier to maintain and less prone to forgetting adding/updating the second line (also see similar cases below

Reply:
Done